root_orchestrator/system-manager-python/tests/services_test.py

```diff
@@ -1,278 +1,268 @@
-    app_id = apps_db.mongo_add_application(db_app_mock["applications"][0])
-    sla["applications"][0]["applicationID"] = app_id
-
-    sla["applications"][0]["microservices"][0]["microservice_name"] = "TOOLONGNAME"
-
-    # EXEC
-    result, code = create_services_of_app("Admin", sla)
-
-    # ASSERT
-    assert code == 403
-
-
-@mongomock.patch(servers=(("localhost", 10007),))
-def test_create_invalid_service_namespace():
-    # SETUP
-    mockdb()
-
-    sla = get_full_random_sla_app()
-    db_app_mock = copy.deepcopy(sla)
-
-    db_app_mock["applications"][0]["userId"] = "Admin"
-    db_app_mock["applications"][0]["microservices"] = []
-    app_id = apps_db.mongo_add_application(db_app_mock["applications"][0])
-    sla["applications"][0]["applicationID"] = app_id
-
-    sla["applications"][0]["microservices"][0][
-        "microservice_namespace"
-    ] = "THISNAMESPACEISTOOLONGTOBECCEPTED"
-
-    # EXEC
-    result, code = create_services_of_app("Admin", sla)
-
-    # ASSERT
-    assert code == 403
-
-
-@mongomock.patch(servers=(("localhost", 10007),))
-def test_delete_service():
-    # SETUP
-    mockdb()
-
-    sla = get_full_random_sla_app()
-    db_app_mock = copy.deepcopy(sla)
-
-    db_app_mock["applications"][0]["userId"] = "Admin"
-    db_app_mock["applications"][0]["microservices"] = []
-    app_id = apps_db.mongo_add_application(db_app_mock["applications"][0])
-    sla["applications"][0]["applicationID"] = app_id
-
-    result, code = create_services_of_app("Admin", sla)
-    db_app_before_deletion = apps_db.mongo_find_app_by_name_and_namespace(
-        sla["applications"][0]["application_name"],
-        sla["applications"][0]["application_namespace"],
-    )
-    service_to_be_deleted = db_app_before_deletion["microservices"][0]
-
-    # EXEC
-    result = delete_service("Admin", service_to_be_deleted)
-    resultNone = delete_service("Admin", service_to_be_deleted)
-
-    # ASSERT
-    assert result
-    assert not resultNone
-    db_app_after_deletion = apps_db.mongo_find_app_by_name_and_namespace(
-        sla["applications"][0]["application_name"],
-        sla["applications"][0]["application_namespace"],
-    )
-    assert service_to_be_deleted not in db_app_after_deletion["microservices"]
-    assert apps_db.mongo_find_job_by_id(service_to_be_deleted) is None
+        created_app = app_operations.create_app("Admin", mock_data["applications"][0])
+        app_id = created_app["applicationID"]
+        sla["applications"][0]["applicationID"] = app_id
+
+        microservices = sla["applications"][0]["microservices"]
+        last_service = microservices[len(microservices) - 1]
+        last_service = generate_db_structure(created_app, last_service)
+
+        # EXEC
+        result, code = create_services_of_app("Admin", sla)
+
+        # ASSERT
+        assert code == 200
+        db_app_result = app_operations.get_app_by_name_and_namespace(
+            sla["applications"][0]["application_name"],
```

Comment:
Nitpick/question-of-preferences:
In general this is written explicitly very often:  sla["applications"][0]
It is fine and all. How about creating a tiny helper function that gets the first app and calling it and when we have this  sla["applications"][0] multiple times in the same context (e.g. same function) fetch the application once and reuse it instead of calling  sla["applications"][0] over and over?

This is up to you if you want to do this or not. (It just feels a bit off to me)

Reply:
Sure thing, it's more readable the way you suggested.